### PR TITLE
tsnet: fix panic caused by logging after test finishes

### PR DESCRIPTION
**Pull request by andrew-d**
Updates #13773

Change-Id: I95e03eb6aef1639bd4a2efd3a415e2c10cdebc5a

### CODE REVIEW

File: tsnet/tsnet_test.go [hunk 1]
```diff
@@ -510,7 +510,7 @@ func TestStartStopStartGetsSameIP(t *testing.T) {
 			Dir:        tmps1,
 			ControlURL: controlURL,
 			Hostname:   "s1",
-			Logf:       logger.TestLogger(t),
+			Logf:       tstest.WhileTestRunningLogger(t),
```
Comment (creachadair) [inline tsnet/tsnet_test.go:513]: Could we not also achieve the same thing by waiting for close rather than deferring it at the end?

**Review by creachadair — APPROVED** [review]

File: tsnet/tsnet_test.go (hunk 1 above)
Comment (nickkhyl) [reply tsnet/tsnet_test.go:513]:
```
panic: Log in goroutine after TestStartStopStartGetsSameIP has completed:
 ... health(warnable=no-derp-connection): error: Tailscale could not connect to the relay server with ID '0'. Your Internet connection might be down, or the server might be temporarily unavailable.
goroutine 14667 [running]:
testing.(*common).logDepth(0xc000354d00, {0xc012ae0000, 0xc8}, 0x3)
        /snap/go/10730/src/testing/testing.go:1029 +0x4c5
testing.(*common).log(...)
        /snap/go/10730/src/testing/testing.go:1011
testing.(*common).Logf(0xc000354d00, {0xc012a74000?, 0x185c661?}, {0xc01020c020?, 0x1885c0a?, 0x1e?})
        /snap/go/10730/src/testing/testing.go:1062 +0x54
tailscale.com/tsnet.TestStartStopStartGetsSameIP.TestStartStopStartGetsSameIP.func1.TestLogger.func2({0x1885c0a, 0x1e}, {0xc01020c020, 0x2, 0x2})
        /types/logger/logger.go:394 +0x94      
tailscale.com/tsnet.(*Server).start.func1({0x1885c0a?, 0x94?}, {0xc01020c020?, 0x16?, 0x417845?})
        /tsnet/tsnet.go:517 +0x96
tailscale.com/ipn/ipnlocal.(*LocalBackend).onHealthChange(0xc00081c008, 0xc0002fe700, 0xc01545c000)
        /ipn/ipnlocal/local.go:845 +0x109      
created by tailscale.com/health.(*Tracker).setUnhealthyLocked.func1 in goroutine 14666
        /health/health.go:402 +0x13a
FAIL    tailscale.com/tsnet     13.126s
FAIL
```
The logging (and panic) occurs in a fire-and-forget goroutine:
https://github.com/tailscale/tailscale/blob/52ef27ab7c3078513ea7e265ce4a1b3e0adea833/health/health.go#L397-L404

Comment (creachadair) [reply tsnet/tsnet_test.go:513]: That's a regrettable choice.

**Review by bradfitz — APPROVED** [review]

File: tsnet/tsnet_test.go (hunk 1 above)
Comment (bradfitz) [reply tsnet/tsnet_test.go:513]: It used to be worse. :)

**Comment by bradfitz** [comment]
This is worth a tracking bug to both fix the fire-and-forget goroutine and audit all other tests that don't use this logging wrapper. Seems like an Effort, per go/cleanup.

**Pull request merged**